File: frontend/src/api/audit.ts
import { request } from './index'

export interface AuditEntry {
  id: number
  timestamp: string
  operator: string
  action: string
  target_type: string
  target_id: string
  summary_before: Record<string, unknown> | null
  summary_after: Record<string, unknown> | null
}

export interface AuditQueryResult {
  items: AuditEntry[]
  total: number
}

export interface AuditQueryParams {
  page?: number
  limit?: number
  action?: string
  target_type?: string
  operator?: string
  since?: string
  until?: string
}

export async function fetchAuditEntries(params: AuditQueryParams = {}): Promise<AuditQueryResult> {
  const query = new URLSearchParams()
  if (params.page) query.set('page', String(params.page))
  if (params.limit) query.set('limit', String(params.limit))
  if (params.action) query.set('action', params.action)
  if (params.target_type) query.set('target_type', params.target_type)
  if (params.operator) query.set('operator', params.operator)
  if (params.since) query.set('since', params.since)
  if (params.until) query.set('until', params.until)

  const qs = query.toString()
  return request(`/api/audit${qs ? '?' + qs : ''}`)
}

File: src/quickquip/app/web/routes/audit.py
from fastapi import APIRouter, Query

from quickquip.app.web.audit import audit_logger

router = APIRouter()


@router.get("/audit")
def get_audit_entries(
    page: int = Query(default=1, ge=1),
    limit: int = Query(default=50, ge=1, le=200),
    action: str | None = Query(default=None),
    target_type: str | None = Query(default=None),
    operator: str | None = Query(default=None),
    since: str | None = Query(default=None),
    until: str | None = Query(default=None),
):
    items, total = audit_logger.query(
        page=page,
        limit=limit,
        action=action,
        target_type=target_type,
        operator=operator,
        since=since,
        until=until,
    )
    return {"items": items, "total": total}
